redis/deployment/src/main/java/io/quarkiverse/langchain4j/redis/RedisEmbeddingStoreProcessor.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkiverse.langchain4j.redis;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkiverse.langchain4j.redis.runtime.RedisEmbeddingStoreConfig;
+import io.quarkiverse.langchain4j.redis.runtime.RedisEmbeddingStoreRecorder;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.redis.deployment.client.RequestedRedisClientBuildItem;
+
+public class RedisEmbeddingStoreProcessor {
+
+    public static final DotName REDIS_EMBEDDING_STORE = DotName.createSimple(RedisEmbeddingStore.class);
+
+    private static final String FEATURE = "langchain4j-redis";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    public RequestedRedisClientBuildItem requestRedisClient(RedisEmbeddingStoreConfig config) {
+        return new RequestedRedisClientBuildItem(config.redisClientName());
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void createBean(
+            BuildProducer<SyntheticBeanBuildItem> beanProducer,
+            RedisEmbeddingStoreRecorder recorder,
+            RedisEmbeddingStoreConfig config) {
+        beanProducer.produce(SyntheticBeanBuildItem
+                .configure(REDIS_EMBEDDING_STORE)
+                .setRuntimeInit()
+                .defaultBean()
+                .scope(ApplicationScoped.class)
+                .supplier(recorder.embeddingStoreSupplier(config))
```

Comment:
would need to be a function (context -> ....) and get the injection point declared here.